LBHTenancyAPI/UseCases/V2/Search/SearchTenancyUseCase.cs

```diff
@@ -72,22 +66,49 @@ public async Task<SearchTenancyResponse> ExecuteAsync(SearchTenancyRequest reque
             return useCaseResponse;
         }
 
-        private void updateUniqueListItems(List<TenancyListItem> uniqueTenancyListItems,
-            IEnumerable<IGrouping<string, TenancyListItem>> groupsOfDuplicatesTenancyListItems)
+        private static List<TenancyListItem> MergeJointTenancies(List<TenancyListItem> allTenancyListItems)
         {
-            foreach (var grouping in groupsOfDuplicatesTenancyListItems)
+            var duplicateTenancies = new List<List<TenancyListItem>>();
+            var newResult = new List<TenancyListItem>();
+            foreach (var tenancy in allTenancyListItems)
+            {
+                var duplicateTenancy = allTenancyListItems.Find(
+                    x => x.TenancyRef == tenancy.TenancyRef &&
+                         x.PrimaryContactName != tenancy.PrimaryContactName
+                );
+                if (duplicateTenancy.TenancyRef != null)
+                {
+                    duplicateTenancies.Add(new List<TenancyListItem> {tenancy, duplicateTenancy});
+                    newResult.Remove(duplicateTenancy);
+                }
+                else
+                {
+                    newResult.Add(tenancy);
+                }
+            }
+
+            if (duplicateTenancies.Count <= 0) return newResult;
```

Comment:
Why don't you have an `Empty()`, C# :(

We could use this here:
```suggestion
            if (!duplicateTenancies.Any()) return newResult;
```

But that `!` is fairly quiet, so you may not like this more.